File: src/main/kotlin/hanablive/entity/dto/ChatMessage.kt
package hanablive.entity.dto

import java.time.ZonedDateTime

data class ChatMessage(
    val msg: String,
    val who: String,
    val discord: Boolean,
    val server: Boolean,
    val datetime: ZonedDateTime,
    val room: String,
    val recipient: String,
)

File: src/external/hanablive/dto/chat_message.rs
use chrono::{DateTime, Utc};
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ChatMessage {
    pub msg: String,
    pub who: String,
    pub discord: bool,
    pub server: bool,
    pub datetime: DateTime<Utc>,
    pub room: String,
    pub recipient: String,
}
